project_task_analytic_propagation/models/account_analytic_line.py

```diff
@@ -7,18 +7,41 @@ class AccountAnalyticLine(models.Model):
     _inherit = "account.analytic.line"
 
     account_id = fields.Many2one(
-        compute="_compute_account_id",
-        store=True,
+        "account.analytic.account",
+        "Project Account",
+        ondelete="restrict",
+        index=True,
+        check_company=True,
+        compute="_compute_analytic_accounts",
         readonly=False,
+        store=True,
     )
 
-    @api.depends("task_id.analytic_account_id")
-    def _compute_account_id(self):
-        """Change the analytic account non billed lines."""
+    def _get_analytic_plan_columns(self):
+        return [
+            f"task_id.{x}"
+            for x in self.task_id._get_plan_fnames()
+            if self._fields.get(x)
+        ]
+
+    @api.depends(lambda x: x._get_analytic_plan_columns())
```

Comment:
this makes it clearer what `x` is:

```suggestion
    @api.depends(lambda self: self._get_analytic_plan_columns())
```